webapp-django/crashstats/supersearch/views.py

```diff
@@ -97,6 +98,12 @@ def get_params(request):
     return params
 
 
+@ratelimit(
+    key='ip',
+    rate=utils.ratelimit_rate,
+    method=ratelimit.ALL,
+    block=True
+)
 @pass_default_context
 def search(request, default_context=None):
```

Comment:
To make this statement a bit clearer: 

loading ``/search/`` gets the page layout, all the JS code and stuff, and then it does an AJAX call to ``/search/results/``. That one actually calls the middleware and Elasticsearch. So, as is this patch, if someone where to use a crawler to load ``/search/`` many times, we would be protected. However, if that evil person calls ``/search/results/`` instead, it would have the same effect and we would not be protected. The same goes for someone that loads the UI in a browser and just spams the Search button.